cmake/interrogate_setup_dot_py.py

```diff
@@ -117,7 +117,7 @@ def generate_cmake_file(package_name, version, scripts, package_dir, pkgs, modul
             continue
         # check every child has the same root folder as its parent
         parent_name = '.'.join(splits[:1])
-        if location != locations[parent_name]:
+        if not locations[parent_name] in location:
```

Comment:
I guess this is supposed to check if the parent location is actually a prefix of the location? Currently this could match anywhere within the string - not only at the start.